src/data/pi.lean

```diff
@@ -144,7 +144,73 @@ function.update_injective _ i
 @[simp] lemma single_inj (i : I) {x y : f i} : pi.single i x = pi.single i y ↔ x = y :=
 (pi.single_injective _ _).eq_iff
 
+end single
+
+section single1 -- A copy of the pi.single API for `has_one`
+
+variables [decidable_eq I]
+variables [Π i, has_one (f i)] [Π i, has_one (g i)] [Π i, has_one (h i)]
+
+/-- The function supported at `i`, with value `x` there, and 1 elsewhere. -/
+def single1 (i : I) (x : f i) : Π i, f i :=
+function.update 1 i x
```

Comment:
I’d probably break this section out into a separate PR, unless someone tells me that this isn’t how it should be done anyways.

Reply:
Joachim Breitner: In the end I got this: https://github.com/leanprover-community/mathlib/pull/11825/files
I had to define pi.single1 as a variant of pi.single that defaults to 1 (not 0). I should probably break that out into a separate PR. What would be a good name for it?

Yaël Dillies: What seems reasonable to me is to rename pi.single to pi.add_single and pi.single1 to pi.single, or to just rename pi.single1 to pi.mul_single.

Yaël Dillies: Either way, you shouldn't have to write an entire new API. Instead, you should refactor the existing API to talk about multiplicative stuff and add @[to_additive] on all declarations to automatically generate the additive version.

Reply:
This can go once we have https://github.com/leanprover-community/mathlib/pull/11849